security/security-core/src/main/java/io/camunda/security/auth/Authentication.java

```diff
@@ -31,12 +32,14 @@ public final class Authentication {
   @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
   public Authentication(
       final @JsonProperty("authenticated_username") String authenticatedUsername,
+      final @JsonProperty("authenticated_application_id") String authenticationApplicationId,
```

Comment:
:wrench: Could we make those two mutually exclusive? It's either username or application id, never both, right?

Reply:
I'm not completely sure I'm following but I think thats likely on my side :) - they are by their nature exclusive, its either a user token or its a M2M token, but I think in the future we'd want to possible distinguish between the two down the chain so I wouldn't opt to collapse the fields